### PR TITLE
Use global for select2 args and set width to 100%

### PR DESCRIPTION
**Pull request by jom**
This fixes an issue with window resizing and renames the `job_manager_select2_multiselect_args` variable to more generic `job_manager_select2_args`.

To fix the issue with window resizing, it passes a more opinionated 100% width. Seems to look okay in the themes I tested with.

### Testing Instructions
- On multiple themes, check:
   - `[jobs]` filter with multiselect and normal select for the job categories.
   - `[submit_job_listing]` page with multiselect fields.

These should all be `select2` unless `chosen` has already loaded from the theme or the theme is explicitly blocking it.

### CODE REVIEW

File: assets/js/ajax-filters.js [hunk 1]
```diff
@@ -239,14 +239,11 @@ jQuery( document ).ready( function ( $ ) {
 		return false;
 	} );
 
-	if ( $.isFunction( $.fn.select2 ) ) {
-		var select2_args = {
-			allowClear: true,
-			minimumResultsForSearch: 10
-		};
-		if ( 1 === parseInt( job_manager_ajax_filters.is_rtl, 10 ) ) {
```
Comment (donnapep) [inline assets/js/ajax-filters.js]: Just wondering is this was intentionally removed?

File: assets/js/multiselect.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 /* global job_manager_select2_args */
-jQuery(function(){
-	jQuery( '.job-manager-multiselect:visible' ).select2( job_manager_select2_args );
+jQuery(function($){
+	if ( $.isFunction( $.fn.select2 ) && typeof job_manager_select2_args !== 'undefined' ) {
+		$('.job-manager-multiselect:visible').select2(job_manager_select2_args);
```
Comment (donnapep) [inline assets/js/multiselect.js]: We've lost the lovely whitespacing before and after the parentheses. 🙂

File: assets/js/term-multiselect.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 /* global job_manager_select2_args */
-jQuery(function(){
-	jQuery( '.job-manager-category-dropdown:visible' ).select2( job_manager_select2_args );
+jQuery(function($){
+	if ( $.isFunction( $.fn.select2 ) && typeof job_manager_select2_args !== 'undefined' ) {
+		$('.job-manager-category-dropdown:visible').select2(job_manager_select2_args);
```
Comment (donnapep) [inline assets/js/term-multiselect.js]: Same here.

File: assets/js/ajax-filters.js (hunk 1 above)
Comment (jom) [reply assets/js/ajax-filters.js]: That is now coming in from `job_manager_select2_args`.

File: wp-job-manager.php [hunk 4]
```diff
@@ -410,9 +410,11 @@ public function frontend_scripts() {
 				$select2_args['dir'] = 'rtl';
 			}
 
+			$select2_args['width'] = '100%';
+
 			wp_localize_script(
-				'select2', 'job_manager_select2_multiselect_args',
-				apply_filters( 'job_manager_select2_multiselect_args', $select2_args )
```
Comment (alexsanford) [inline wp-job-manager.php]: Do we need to deprecate this? Or has it not been released yet?

Comment (jom) [reply wp-job-manager.php]: It hasn't been released yet. All `select2` stuff is in this release.

**Pull request merged**